src/parser/postgresparser.cpp

```diff
@@ -800,6 +800,143 @@ parser::SQLStatement* PostgresParser::CreateTransform(CreateStmt* root) {
+}
+
+// This helper function takes in a Postgres TypeName object and transforms
+// it into a Peloton ReturnType object
+parser::ReturnType* PostgresParser::ReturnTypeTransform(TypeName* root) {
+  parser::ReturnType::DataType data_type;
+  char* name = (reinterpret_cast<value*>(root->names->tail->data.ptr_value)
+                    ->val.str);
+  parser::ReturnType* result = nullptr;
+
+  // Transform parameter type
+  if ((strcmp(name, "int") == 0) || (strcmp(name, "int4") == 0)) {
+    data_type = FuncParameter::DataType::INT;
+  } 
+  //*******See which of the below are needed later***********
+  // I think text/varchar may be used for the pg_catalog
+  else if (strcmp(name, "varchar") == 0) {
+    data_type = FuncParameter::DataType::VARCHAR;
+  } else if (strcmp(name, "int8") == 0) {
+    data_type = FuncParameter::DataType::BIGINT;
+  } else if (strcmp(name, "int2") == 0) {
+    data_type = FuncParameter::DataType::SMALLINT;
+  } else if (strcmp(name, "text") == 0) {
+    data_type = FuncParameter::DataType::TEXT;
+  } else if (strcmp(name, "tinyint") == 0) {
+    data_type = FuncParameter::DataType::TINYINT;
+  } else if(strcmp(name, "bool") == 0) {
+     data_type = FuncParameter::DataType::BOOL;
+  } else {
+    LOG_ERROR("Column DataType %s not supported yet...\n", name);
+    throw NotImplementedException("...");
+  }
+
+  // Transform Varchar parameter name
+  result = new ReturnType(data_type);
+
+  return result;
+}
+
+// This function takes in a Postgres CreateFunctionStmt parsenode
+// and transfers into a Peloton CreateFunctionStatement parsenode.
+// Please refer to parser/parsenode.h for the definition of
+// CreateFunctionStmt parsenodes.
+parser::SQLStatement* PostgresParser::CreateFunctionTransform(CreateFunctionStmt* root) {
+  UNUSED_ATTRIBUTE CreateFunctionStmt* temp = root;
+  parser::CreateFunctionStatement* result = new CreateFunctionStatement();
+   
+  result->replace = root->replace;
+  //FunctionParameter* parameters = root->parameters;
+
+  result->func_parameters = new std::vector<FuncParameter*>();
+  for (auto cell = root->parameters->head; cell != nullptr; cell = cell->next) {
+    Node* node = reinterpret_cast<Node*>(cell->data.ptr_value);
+    if ((node->type) == T_FunctionParameter) {
+      // Transform Function Parameter
+      FuncParameter* funcpar_temp =
+          FunctionParameterTransform(reinterpret_cast<FunctionParameter*>(node));
+      
+      result->func_parameters->push_back(funcpar_temp);
+    }
+  }
+ 
+  ReturnType* ret_temp =
+          ReturnTypeTransform(reinterpret_cast<TypeName*>(root->returnType));
+  result->return_type = ret_temp;
+
+  // Assuming only one function name can be passed for now.
+  char* name = (reinterpret_cast<value*>(root->funcname->tail->data.ptr_value)
+                    ->val.str);
+  std::string func_name_string(name);
+  result->function_name = func_name_string; //We may have to change this to another function for handlign char * to string
```

Comment:
@xinzhang1234 Will do. Wasn't aware the = operator for string was overloaded to handle char* types too.